### ISSUE TITLE
Introducing GoogleTest for unit tests

### ISSUE THREAD

**Issue by btemuer** [exadg/exadg#359]
I wish to raise the question if we'd like to introduce a testing framework like [GoogleTest](https://github.com/google/googletest) for writing unit tests.

Right now we only have a few unit tests that rely on our own implementation which is primitive compared to what a testing framework can offer. Having GoogleTest can help us write more and better tests to test any specific part of the code and become more resistant to bugs and errors.

I'm excited to hear your thoughts and opinions!

**Comment by peterrum** [comment]
Good idea :+1: Let's give it a try. When you have something, we could take a look at the pros/cons of gtest/ctest.

**Comment by btemuer** [comment]
I just created the PR #360, introducing Googletest where I have written two tests with it. You can have a look at a04ce51c39826cf0cf2e686ccceab9de6aee5758 to see the changes 😄
